flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBHandle.java

```diff
@@ -202,12 +202,21 @@ private void restoreInstanceDirectoryFromPath(Path source) throws IOException {
             final String fileName = file.getFileName().toString();
             final Path targetFile = instanceRocksDBDirectory.resolve(fileName);
             if (fileName.endsWith(SST_FILE_SUFFIX)) {
-                // hardlink'ing the immutable sst-files.
-                Files.createLink(targetFile, file);
-            } else {
-                // true copy for all other files.
-                Files.copy(file, targetFile, StandardCopyOption.REPLACE_EXISTING);
+                try {
+                    // hardlink'ing the immutable sst-files.
+                    Files.createLink(targetFile, file);
+                    continue;
+                } catch (IOException ioe) {
+                    logger.debug(
```

Comment:
Why not leave it as `INFO` level?

Reply:
I thought that this would be too verbose and something a user might not be really interested in.

Reply:
But maybe you are right that this is something a user would like to see in order to understand performance differences. I'll change it to `INFO`.